### PR TITLE
fix(tanka single-binary): remove unnecessary PersistentVolumeClaim

### LINKED ISSUE #1245: [tanka single-binary] Remove unnecessary PVC from deployment
**Describe the bug**

There is an extra `PersistentVolumeClaim` which is not needed when deploying Tempo with Tanka as single-binary, as you can see [here](https://github.com/grafana/tempo/blob/5a6a5d95a5907543aa7a63d1b280f1e08aa28ccb/operations/jsonnet/single-binary/tempo.libsonnet#L21).

**To Reproduce**
1. Deploy Tempo with Tanka single-binary
2. Check deployed resources

**Expected behavior**

There should be no extra PersistentVolumeClaim since it's already created by the StatefulSet's persistentVolumeClaimTemplate.

**Environment:**
 - Infrastructure: [e.g., Kubernetes, bare-metal, laptop]
 - Deployment tool: [e.g., helm, jsonnet]

**Additional Context**

### PR DESCRIPTION
**Pull request by irizzant**
**What this PR does**:
Removes unnecessary PersistentVolumeClaim

**Which issue(s) this PR fixes**:
Fixes #1245 

**Checklist**
- [X] Tests updated
- [X] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,7 @@
 * [ENHANCEMENT] Update compactor ring to automatically forget unhealthy entries [#1178](https://github.com/grafana/tempo/pull/1178) (@mdisibio)
 * [ENHANCEMENT] Added the ability to pass ISO8601 date/times for start/end date to tempo-cli query api search [#1208](https://github.com/grafana/tempo/pull/1208) (@joe-elliott)
 * [ENHANCEMENT] Prevent writes to large traces even after flushing to disk [#1199](https://github.com/grafana/tempo/pull/1199) (@mdisibio)
+* [BUGFIX]: Remove unnecessary PersistentVolumeClaim [#1245](https://github.com/grafana/tempo/issues/1245)
```
Comment (mapno) [inline CHANGELOG.md:37]: Can you move this to `main / unreleased`?

**Review by mapno — COMMENTED** [review]
Nice finding, thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (irizzant) [reply CHANGELOG.md]: done as requested @mapno

**Comment by irizzant** [comment]
@mapno you're welcome

**Review by mapno — APPROVED** [review]

**Pull request merged**